src/components/NotificationContext/NotificationContext.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { SnackbarSuccess } from '../Snackbars/SnackbarSuccess';
+import { SnackbarError } from '../Snackbars/SnackbarError';
+
+type NotificationContextType = {
+  notification: { type: string; message: string };
+  setNotification: React.Dispatch<React.SetStateAction<{ type: string; message: string }>> | undefined;
+};
+
+const NotificationContext = React.createContext<NotificationContextType>({
```

Comment:
Why `setNotification` can be undefined? I think it should be typed as
```ts
  React.createContext<NotificationContextType  | undefined>(undefined)
```

Then you don't need to check if(setNotification) is defined in components?